### ISSUE TITLE
Module Dependency violation isn't caught: function on prototype

### ISSUE THREAD

**Issue by svasilinets** [google/closure-compiler#895]
I have following scenario: 
I have STRICT_MODULE_DEP_QNAME.level = CheckLevel.WARNING and following testCode:

```java
  public void testBadModuleDep2() {
    testSame(createModuleStar(
        NAMES,
        "/** @constructor */ var WI = function(){};  WI.prototype.x = function(){};  a.t = new WI();",
        "a.t.x();"
    ), STRICT_MODULE_DEP_QNAME);
  }
```
And I don't see any warnings/errors. 

When I remove call of function x:

```java
  public void testBadModuleDep2() {
    testSame(createModuleStar(
        NAMES,
        "/** @constructor */ var WI = function(){};  WI.prototype.x = function(){};  a.t = new WI();",
        "a.t;"
    ), STRICT_MODULE_DEP_QNAME);
  }
```
I see as expected:
JSC_STRICT_MODULE_DEP_QNAME. cannot reference a.t because of a missing module dependency

**Comment by concavelenz** [comment]
interesting.

**Comment by aslushnikov** [comment]
Any updates on this?

Chrome DevTools currently employ some hacks to workaround this, which makes compilation last for ~4 minutes on top-notch macbook pro.
